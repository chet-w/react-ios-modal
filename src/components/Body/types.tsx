export interface BodyProps {
  children?: JSX.Element[] | JSX.Element | string;
}
