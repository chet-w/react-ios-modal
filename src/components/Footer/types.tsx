export interface FooterProps {
  options: FooterOptions;
}

export type FooterOptions = [FooterOption, FooterOption, FooterOption];

export interface FooterOption
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  content: string;
}
