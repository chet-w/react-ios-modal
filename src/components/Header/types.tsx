export interface HeaderProps {
  title: string;
  onClose?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
