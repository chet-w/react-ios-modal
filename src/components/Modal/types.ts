import { ReactElement } from "react";

export interface ModalProps {
  title: string;
  children: ReactElement;
  closable?: boolean
  onClose?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  clickOutsideToClose?: boolean;
}
