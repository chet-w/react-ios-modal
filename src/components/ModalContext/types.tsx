export interface ModalContextProps {
  isOpen: boolean;
  openModal: (modal: JSX.Element) => void;
  closeModal: () => void;
}

export interface ModalProviderProps {
  children?: JSX.Element;
}
