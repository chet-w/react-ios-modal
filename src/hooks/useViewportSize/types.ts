export type DeviceSize = "xs" | "sm" | "md" | "lg";

export interface ViewportSizeData {
  deviceSize: DeviceSize;
  width: number;
  isMobile: boolean;
};