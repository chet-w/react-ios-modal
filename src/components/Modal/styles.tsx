import styled from "styled-components";
import { motion } from "framer-motion";

export const Modal = styled(motion.section)`
  width: 720px;
  min-height: 75vh;
  background: white;
  box-shadow: 0 0 10px 10px rgba(0, 0, 0, 0.1);
  margin: 12.5vh auto 0 auto;
  border-radius: 16px;
  padding: 0;
  display: flex;
  flex-direction: column;

  @media (max-width: ${(props) => props.theme.breakpoints.sm}px) {
    height: calc(100vh - 32px);
    margin: 32px auto 0 auto;
  }
`;

export const variants = {
  open: {
    y: 0,
    opacity: 1
  },
  closed: {
    y: "100%",
    opacity: 0
  }
};

export const transition = {
  damping: 20,
  restDelta: 0.001,
  stiffness: 180,
  type: "spring"
};
