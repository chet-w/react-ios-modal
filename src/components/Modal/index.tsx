import * as S from "./styles";
import { ModalProps } from "./types";

import Header from "../Header";
import Body from "../Body";
import Footer from "../Footer";

const Modal = (props: ModalProps) => {
  return (
    <S.Modal
      variants={S.variants}
      transition={S.transition}
      initial="closed"
      animate="open"
      exit="closed"
    >
      <Header
        title={props.title}
        closable={props.closable}
        onClose={props.onClose}
      />
      <Body>{props.children}</Body>
      {props.footerOptions && <Footer options={props.footerOptions} />}
    </S.Modal>
  );
};

export default Modal;
