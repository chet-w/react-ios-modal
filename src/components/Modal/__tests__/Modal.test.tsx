import { render } from "@testing-library/react";
import Modal from "../";
import { ModalProps } from "../types";
import WithTheme from "../../../test-utilities/WithTheme";

const renderModal = (props?: ModalProps) => {
  props = props || { id: "test", title: "test" };
  const { container } = render(
    <WithTheme>
      <Modal {...props} />
    </WithTheme>
  );
  return container;
};

describe("Modal", () => {
  it("renders successfully", () => {
    const modal = renderModal();
    expect(modal).toBeTruthy();
  });

  it("has the correct mobile styling", () => {
    const background = renderModal().firstElementChild as HTMLElement;
    const styles = getComputedStyle(background);
    expect(styles).toMatchSnapshot();
  });
});
