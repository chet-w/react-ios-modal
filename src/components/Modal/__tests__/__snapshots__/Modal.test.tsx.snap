// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal has the correct mobile styling 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "width",
  "10": "visibility",
  "11": "opacity",
  "12": "transform",
  "13": "user-select",
  "14": "transform-origin",
  "2": "min-height",
  "3": "background",
  "4": "box-shadow",
  "5": "margin",
  "6": "border-radius",
  "7": "padding",
  "8": "flex-direction",
  "9": "max-width",
  "_importants": Object {
    "background": "",
    "border-radius": "",
    "box-shadow": "",
    "display": "",
    "flex-direction": "",
    "margin": "",
    "max-width": "",
    "min-height": "",
    "opacity": "",
    "padding": "",
    "transform": "",
    "transform-origin": "",
    "user-select": "",
    "visibility": undefined,
    "width": "",
  },
  "_length": 15,
  "_onChange": [Function],
  "_values": Object {
    "background": "white",
    "background-color": "white",
    "border-radius": "16px",
    "box-shadow": "0 0 10px 10px rgba(0,0,0,0.1)",
    "display": "flex",
    "flex-direction": "column",
    "margin": "12.5vh auto 0px auto",
    "margin-bottom": "0px",
    "margin-left": "auto",
    "margin-right": "auto",
    "margin-top": "12.5vh",
    "max-width": "100%",
    "min-height": "75vh",
    "opacity": "0",
    "padding": "0px",
    "padding-bottom": "0px",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "0px",
    "transform": "translate3d(0px, 400px, 0) scale(NaN, NaN)",
    "transform-origin": "50% 50% 0",
    "user-select": "none",
    "visibility": "visible",
    "width": "720px",
  },
}
`;
